dhcpv6/dhcpv6message.go

```diff
@@ -42,8 +41,9 @@ func GetTime() uint32 {
 	return uint32((now.Nanoseconds() / 1000000000) % 0xffffffff)
 }
 
-// NewSolicitWithCID creates a new SOLICIT message with CID.
-func NewSolicitWithCID(duid Duid, modifiers ...Modifier) (*Message, error) {
+// NewSolicit creates a new SOLICIT message, using the given hardware address to
+// derive the IAID in the IA_NA option.
+func NewSolicit(duid Duid, hwaddr net.HardwareAddr, modifiers ...Modifier) (*Message, error) {
```

Comment:
What about removing the duid parameter and generating a default one?

Reply:
Based on the hwaddr

Reply:
yeah, DUID_LLT generated with GetTime and MAC address, sounds good